src/assembly/types.ts

```diff
@@ -0,0 +1,45 @@
+import { ConstructInfo } from 'aws-cdk-lib/core/lib/private/runtime-info';
+import { DockerImageManifestEntry, FileManifestEntry } from 'cdk-assets';
+
+export type StackAsset = FileManifestEntry | DockerImageManifestEntry;
+
+/**
+ * Map of CDK construct path to logicalId
+ */
+export type StackMetadata = { [path: string]: string };
+
+/**
+ * ConstructTree is a tree of the current CDK construct
+ * It represents the structure in the `tree.json` file and is based
+ * off the implementation here:
+ * https://github.com/aws/aws-cdk/blob/4bce941fc680ebd396569383f6cf07527541dcc2/packages/aws-cdk-lib/core/lib/private/tree-metadata.ts?plain=1#L177
```

Comment:
Presumably no aws-cdk code can be reused instead, since it's private?

Reply:
For some reason I thought it wasn't exported, but it is. Updated to use the cdk import.